module/Finna/src/Finna/Cover/Loader.php

```diff
@@ -397,18 +411,52 @@ protected function processImageURL($url, $cache = true)
             return false;
         }
 
+        if (!$this->getUnsizedImage($url)) {
+            return false;
+        }
+        // Set local file path to temporary file if the image is not to be cached
+        $targetFile = $cache ? $this->localFile : str_replace('.jpg', uniqid(), $this->localFile) . '.jpg';
+        // If the requested image has width and height of 0, then return the unsized image
+        if (!$this->width && !$this->height) {
+            $this->localFile = $this->unsizedImageFile;
+        } elseif (!$this->createResizedImage($this->unsizedImageFile, $targetFile)) {
+            return false;
+        }
+
+        if (!$this->localFile) {
```

Comment:
Onko tämä tarkistus oikeassa kohdassa, kun `$this->localFile` käytetään jo ylempänä?